libobs/obs-display.c

```diff
@@ -19,12 +19,32 @@
 #include "obs.h"
 #include "obs-internal.h"
 
+#if defined(_WIN32)
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN 1
+#endif
+#include <Windows.h>
+#endif
+
 bool obs_display_init(struct obs_display *display,
 		      const struct gs_init_data *graphics_data)
 {
 	pthread_mutex_init_value(&display->draw_callbacks_mutex);
 	pthread_mutex_init_value(&display->draw_info_mutex);
 
+#if defined(_WIN32)
+	/* Conservative test for NVIDIA flickering on Intel display */
+	SYSTEM_POWER_STATUS status;
+	display->use_clear_workaround =
+		!GetSystemPowerStatus(&status) ||
+		(status.BatteryFlag != 128 && gs_get_adapter_count() != 1);
```

Comment:
This would also affect Intel+AMD or AMD+AMD laptops or any multi-adapter system with a battery, wouldn't it? Are we able to check the Vendor ID of the adapter at this point? From my testing, this only occurs when OBS is running on the NVIDIA GPU/adapter. Just wondering if we can narrow the field a bit.

Reply:
I don't want to add to the public interface, nor add a d3d12 lib dependency to libobs for a temp hack. The perf hit isn't horrendous; I just don't want it to affect me. :)

Reply:
And this is still tighter than what we have now, which hits everyone.

Reply:
Actually, maybe you should try your laptop on Linux to see if their Linux driver is also affected.

Reply:
The flicker does not seem to occur on Linux with proprietary NVIDIA drivers.